### ISSUE TITLE
Airmode activation via switch

### ISSUE THREAD

**Issue by rauar** [flyduino/kissfc-firmware#11]
TLDR: Kiss FC is definitely missing a manual switch for Airmode activation. The current way of Airmode activation via Min Command is a) dangerous and b) not simple at all (although it easily could be made simple by assigning one of the AUX ports to control Airmode).

> As the KISS FC manual describes:
> 
> If Min Command is set lower than TX Throttle cut, the PIC controller won’t be shut off
> If Min Command is set to 1000 and the Arm function is assigned to a switch on your radio, the Air Mode is
> activated automatically. In Airmode on the Ground you might find the FC trying to regulate “something”
> though it is sitting still. However the Airmode only makes sense in flight.

Notice the last sentence: there's no point in having Airmode active on start or land.

Short background: Airmode actives the PID controller. That means when Airmode is inactive the flight controller will be very unstable.

 I've tested Airmode (Minmode) in combination with throttle cut (Min Throttle) in 2 scenarios:

- Starting
- Rolls/Loops

The current solution using Min Command only works for one of both scenarios but not for both at the same time:

- If Min Command < Min Throttle: Airmode active during flight (good) but also active during start/land (bad)
- If Min Command > Min Throttle: Airmode inactive during start/land (good) but also potentially inactive during freestyle moves (e.g. loops with minimum throttle)

As a conclusion In both scenarios accidents may happen and it's obvious that using Throttle Cut (aka Idle Up) is not a solution at all.

This could be easily improved by allowing Airmode activation via AUX channel. Most people arm their devices anyway using an AUX channel and KISS even supports 3 levels on every AUX channel. 

Therefore the perfect single-switch configuration would be something like:
- AUX 1 Low/Off: Disarmed
- AUX 1 Medium: Armed + Airmode Off
- AUX 2 High: Armed + Airmode On

By this setup one could place a quad on the ground (Disarmed, Aux 1 = Low), then take-off with Airmode off (Armed, Aux 1 = Medium) and after successful start then active Airmode (Aux 1 = High). Same procedure in reverse applies to landing.

I know that it's currently not possible to assign 2 functions to one AUX channel (assuming Airmode is considered a separate function beside Arming). But it could still be made work by using 2 separate AUX channels (either with 2 separate switches on the remote or by using the mixer and assigning the same switch to 2 AUX channels in parallel).

A last remark: I don't understand why Airmode activation is currently more complicated than it needs to be. In the end the pilot should have full control instead of magic software settings (magic = software making assumptions if the device is currently flying or not).

**Comment by tobiasbischoff** [comment]
Hmm why would you want to take off without air mode? I don't see the point honestly. If you just want to run your Motors without any PID active you should have a look into an IDLE- Up Setup (Steele has a detailed Video).

**Comment by rauar** [comment]
> Hmm why would you want to take off without air mode? I don't see the point honestly. 

Airmode is currently activated/deactivated depending on the thrust level. The assumption that the thrust level is a clear indicator if a quad is flying or not is flawed. That's why I'm asking to let the pilot decide.

Airmode during start/land can be nasty if external forces are impacting the quad (e.g. like a slight bounce on the ground). Another thing is steering input from the remote (pitch,yaw) witht the quad still touching ground. I did a test where airmode was active (min command < min throttle) and the quad stil on the ground. If you trigger pitch/yaw just slightly for a fraction of a second and revert to neutral I observed that motors start spinning consistently faster and faster althought sticks are already back in neutral. I think the PID controller is much longer trying to correct because the previously calculated orientation cannot be achieved (because the quad is still on ground).

> If you just want to run your Motors without any PID active you should have a look into an IDLE- Up Setup (Steele has a detailed Video).

I want to have airmode active during flight - but not during start/land. I know Steele's video about Idle Up. Idle Up is simply bumbing up min throttle so that motors don't turn off. It does not impact if the PID controller is active or not as far as I know and therefore is does not solve the problem that the PID controller might misbehave during start/land.
